### PR TITLE
[Ops][Feature] Support DFlash2 speculative decoding on Ascend

### PR DESCRIPTION
**Pull request by yiminghub2024**
[Feat][Spec Decode] Support DFlash2 speculative decoding on Ascend

### What this PR does / why we need it?

Adds DFlash2 speculative decoding on Ascend, and the drafter wiring
GLM-5.3-Flash needs to use it.

DFlash2 is a block-diffusion drafter that ships as its own checkpoint, unlike
MTP which lives in the target checkpoint. Three pieces make it work here: the
drafter is built from the DFlash2 subclasses rather than the base ones, the
platform patch admits the method through vLLM v1's speculative config, and the
GLM model collects the auxiliary hidden states the drafter conditions on.

Two fixes come along because this path reaches them. `AscendLogitsProcessor` did
not accept upstream's `skip_gather`, which the spec-decode logits path passes.
And a draft block is bidirectional, which `full_graph_fia` did not account for:
it decided the FIA sparse mode by asking about the sliding window before asking
whether the batch was causal, so a bidirectional batch was put in band mode
while the mask builder withholds a mask for it -- FIA reads any mask it is given
as defaultMask, which would crop away the very upper triangle the block needs.
The operator then refuses outright:

```
when sparse_mode is 4 (not 0), atten_mask cannot be empty
```

The eager path asked in the other order and was fine; the two now agree.

### Does this PR introduce _any_ user-facing change?

Yes. `--speculative-config` accepts `{"method": "dflash", "model": ...}` on
Ascend. Documented in
`docs/source/user_guide/feature_guide/speculative_decoding.md`.

### How was this patch tested?

Unit tests, 106 across seven files:

```bash
pytest -sv tests/ut/models/test_qwen3_dflash2_subclass_hooks.py \
           tests/ut/models/test_glm5next_aux_hidden_states.py \
           tests/ut/patch/platform/test_patch_v1_dflash2_support.py \
           tests/ut/spec_decode/test_dflash_aux_wiring_source_regression.py \
           tests/ut/spec_decode/test_llm_base_proposer.py \
           tests/ut/ops/test_vocab_parallel_embedding.py \
           tests/ut/attention/test_attention_v1.py
```

On hardware, GLM-5.3-Flash TP8 on Ascend A5, served with

```bash
vllm serve /path/to/GLM-5.3-Flash \
  --speculative-config '{"method":"dflash","model":"/path/to/GLM-5.3-Flash-DFlash2","num_speculative_tokens":7}' \
  --tensor-parallel-size 8 --block-size 640 --max-model-len 65536 \
  --max-num-seqs 16 --no-enable-prefix-caching \
  --compilation-config '{"cudagraph_mode": "FULL_DECODE_ONLY"}'
```

reaching a mean acceptance length of 5.12 out of 7 with coherent output and
decode-FULL graph capture active.

Note that a usable acceptance rate on hardware that falls back to the torch
conv1d path also needs the rewind fix in #PR1_NUMBER. Without it GLM's KDA
layers see only the first draft row, and acceptance is zero past position one.

- vLLM main: https://github.com/vllm-project/vllm/commit/e6bfe03ad73a3330cb427885aa90d97a12e1c704

### CODE REVIEW

File: vllm_ascend/patch/platform/patch_v1_dflash2_support.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    proposer never calls the candidate selector, so the draft would silently
+    degrade to DFlash1. On Ascend the opposite holds: `get_spec_decode_method`
+    routes DFlash2 checkpoints to `AscendDflash2Proposer`, which runs the
+    selector, while the NPU v2 speculator has no DFlash2 path at all.
+    """
+    assert _original_get_v1_unsupported_features is not None
+    unsupported = _original_get_v1_unsupported_features(self)
+    if DFLASH2_UNSUPPORTED_FEATURE in unsupported:
+        unsupported.remove(DFLASH2_UNSUPPORTED_FEATURE)
+    return unsupported
```
[the remark this answers is not in the document]
Comment (yiminghub2024) [reply vllm_ascend/patch/platform/patch_v1_dflash2_support.py]: fixed

File: vllm_ascend/models/glm5next/model.py [hunk 2]
```diff
@@ -56,9 +56,11 @@
     Glm4vForConditionalGeneration,
 )
 from vllm.model_executor.models.interfaces import (
+    EagleModelMixin,
     HasInnerState,
     IsHybrid,
     MixtureOfExperts,
+    SupportsEagle3,
     SupportsPP,
 )
```
[the remark this answers is not in the document]
Comment (yiminghub2024) [reply vllm_ascend/models/glm5next/model.py:65]: Left this as a direct import on purpose. EagleModelMixin and SupportsEagle3 are real bases (Glm5NextModel(nn.Module, EagleModelMixin), Glm5NextForCausalLM(..., SupportsEagle3)), and other patches do isinstance(..., EagleModelMixin). Dummy stand-ins would make those checks succeed without the aux-hidden / Eagle3 methods. MiniMax M3 and DeepSeek V4 already import the same names the same way, and DFlash2 targets vLLM main — a hard ImportError on 0.27.1 is preferable to a silent dummy mixin.

**Comment by yiminghub2024** [comment]
Rebased onto latest ``main`` to fix the merge conflict that was failing ``cpu-ut``.

The CI job rebases onto ``main`` before tests; that step died on ``vllm_ascend/models/qwen3_dflash2.py``. ``main`` already landed the ``decoder_layer_cls`` / ``model_cls`` hooks in #14872, so the conflict was resolved by keeping ``main``'s constructors (no 0.27.1 module-global swap) and retaining this PR's hook unit tests plus the rest of the DFlash2 / GLM-5.3-Flash wiring.
